config_supported.go

```diff
@@ -1,4 +1,14 @@
-//go:build amd64 || arm64
+// Note: The build constraints here are about the compiler, which is more
+// narrow than the architectures supported by the assembler.
+//
+// For example, the internal/platform package has dependencies such as
+// syscall.Mprotect when not on Windows. Go's SDK only supports this on darwin
+// and linux. Constraints here may loosen as the platform package supports
+// other runtime.GOOS, or a plugin becomes available to accomplish the same.
+//
+// Meanwhile, users who know their runtime.GOOS can operate with the compiler
+// may choose to use NewRuntimeConfigCompiler explicitly.
+//go:build (amd64 || arm64) && (darwin || linux || windows)
```

Comment:
Should this line also explicitly reference linux and Darwin instead of not window?

https://github.com/tetratelabs/wazero/blob/main/internal/platform/mmap.go#L1

I think the syscall problem is a compile failure, not a runtime one so given how eager the compiler can be not sure if it wouldn't still try to compile the code referencing Mprotect even on freebsd, even though it's not used at runtime.

Reply:
Sorry think I just repeated @mathetake

Reply:
same general problem area as takeshi's note. yep.